src/components/Graph.react.js

```diff
@@ -109,6 +109,44 @@ class PlotlyGraph extends Component {
         );
     }
 
+    extend(props) {
+        const {id, extendData} = props;
+        var updateData, traceIndices, maxPoints;
+        function getFirstProp(data) {
+            return data[Object.keys(data)[0]];
+        }
+
+        if (typeof extendData === 'object') {
+            if (Array.isArray(extendData)) {
+                [updateData, traceIndices, maxPoints] = extendData;
+            } else {
+                updateData = extendData;
+            }
+
+            if (typeof traceIndices === 'undefined') {
+                function generateIndices(data) {
+                    return Array.from(Array(getFirstProp(data).length).keys());
+                }
+                traceIndices = generateIndices(updateData);
+            }
+
+            if (
+                typeof updateData === 'object' &&
+                Array.isArray(getFirstProp(updateData)) &&
+                Array.isArray(getFirstProp(updateData)[0])
+            ) {
+                return Plotly.extendTraces(
+                    id,
+                    updateData,
+                    traceIndices,
+                    maxPoints
+                );
+            }
```

Comment:
Nice job adapting to missing values 🎉 

Only change I'd ask for is to not fail silently if the wrong values are sent - that would be confusing to debug.

The only things that should get filtered out are falsy values (at least `null` and `undefined` but I don't see any harm filtering everything falsy, ie `if(extendData)` for the outer conditional). Everything else, including this whole inner conditional, should attempt to run. That way plotly.js will throw an error, and the user can see this and figure out how to fix their code.